swift/Lottie.swift

```diff
@@ -0,0 +1,152 @@
+public typealias Buffer = UnsafeMutablePointer<UInt32>
+
+/// Shorthand for a pointer to an internal Lottie animation object
+private typealias Animation = OpaquePointer
+
+/// Shorthand for a pointer to an internal rendering canvas
+private typealias Canvas = OpaquePointer
+
+/// Object used to load and render Lottie frames.
+public class Lottie {
+    /// The number of frames in the Lottie animation.
+    public let numberOfFrames: Int
+
+    /// The duration of the Lottie animation.
+    public let duration: CMTime
+
+    /// The Lottie animation, used for manipulating and rendering frames.
+    private let animation: Animation?
+
+    /// The canvas used for rendering the animation frames, allowing reuse across renders.
+    private var canvas: Canvas? = nil
+
+    /// Initializes a new Lottie instance from a file path.
+    /// - Parameters:
+    ///   - path: The file path of the Lottie animation.
+    public convenience init(path: String) throws {
+        let animation = tvg_animation_new()
+        let picture = tvg_animation_get_picture(animation)
+
+        guard tvg_picture_load(picture, path) == TVG_RESULT_SUCCESS else {
+            throw LottieError.failedToLoadLottieFromPath
+        }
+
+        try self.init(animation: animation)
+    }
+
+    /// Initializes a new Lottie instance from a string representing the animation.
+    /// - Parameters:
+    ///   - string: The string representing the Lottie animation.
+    public convenience init(string: String) throws {
+        let animation = tvg_animation_new()
+        let picture = tvg_animation_get_picture(animation)
+
+        guard let cString = string.cString(using: .utf8),
+              tvg_picture_load_data(picture, cString, UInt32(cString.count), "lottie", "", false) == TVG_RESULT_SUCCESS
+        else {
+            throw LottieError.failedToLoadLottieFromString
+        }
+
+        try self.init(animation: animation)
+    }
+
+    /// Initializes the Lottie instance with frame count and duration extracted from the animation pointer.
+    private init(animation: Animation?) throws {
+        var numberOfFrames: Float = 0
+        tvg_animation_get_total_frame(animation, &numberOfFrames)
+
+        var duration: Float = 0
+        tvg_animation_get_duration(animation, &duration)
+
+        self.animation = animation
+        self.numberOfFrames = Int(numberOfFrames)
+        self.duration = CMTime(seconds: Double(duration), preferredTimescale: 600)
+    }
+
+    /// Renders a Lottie frame into a buffer.
+    /// - Parameters:
+    ///   - index: The index of the frame to render.
+    ///   - buffer: The buffer to render the frame into.
+    ///   - stride: The stride of the buffer.
+    ///   - size: The desired size of the rendered frame.
+    ///   - crop: Optional rectangle to crop the rendered frame.
+    public func render(frameAt index: Int, into buffer: Buffer, stride: Int, size: CGSize, crop: CGRect? = nil) throws {
```

Comment:
> very small thing

maybe `renderFrame(at index: Int` or `render(frameIndex: Int` sounds better?

Reply:
I like the naming of `renderFrame(at index: Int, ...)`. Will update it 😄